cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -590,6 +593,48 @@ boolean isUserTaskDoneExceptionally() {
     }
   }
 
+  // Persist user request and result to log. Called when the user task completes and is moved into COMPLETED status
+  private void persistUserTask(UserTaskInfo userTaskInfo) {
+    String userTaskId = userTaskInfo.userTaskId().toString();
+    String endPoint = userTaskInfo.endPoint().toString();
+    String requestUrl = userTaskInfo.requestWithParams();
+    String startDate = KafkaCruiseControlUtils.toDateString(userTaskInfo.startMs(), DATE_FORMAT, TIME_ZONE);
+    List<OperationFuture> futures = userTaskInfo.futures();
+
+    StringBuilder message = new StringBuilder();
+    message.append(String.format("UserTask [%s] for endpoint [%s] with requestUrl [%s] started-at [%s].",
+                                 userTaskId, endPoint, requestUrl, startDate))
+           .append(String.format(" UserTask has %d step(s).", futures.size()));
+    RESULT_LOG.info(message.toString());
+    message = new StringBuilder();
+
+    int step = 1;
+    for (OperationFuture future : futures) {
+      try {
+        if (future.isDone() || future.isCompletedExceptionally()) {
+          try {
+            AbstractCruiseControlResponse response = ((AbstractCruiseControlResponse) future.get());
+            if (userTaskInfo.parameters() != null) {
```

Comment:
I think `userTaskInfo.parameters() != null` will always be false, can you elaborate a scenario that this will be true?

Reply:
Yeah in UserTaskManagerTest.java, this saves us the effort to mock up `CruiseControlParameters`.